axes/attempts.py

```diff
@@ -60,7 +58,7 @@ def get_cache_key(request_or_obj):
         un = request_or_obj.username
         ua = request_or_obj.user_agent
     else:
-        ip, _ = get_client_ip(request_or_obj)
+        ip = get_client_ip(request_or_obj)
```

Comment:
It would be best if IP is cached on the first request and used on subsequent calls.  Generating cache key should not be more costly than getting the IP.

Reply:
I added caching to a custom request object attribute `axes_request_client_ip` in the `get_client_ip` utility function since the current use of cache simply gets the number of requests from the `axes` cache, and does not account for client IP address. Storing these values into the Django cache would require some more refactoring which is unessential for this change.